app/controllers/admin/content/events.js

```diff
@@ -0,0 +1,133 @@
+          this.notify.success(this.get('l10n').t('Event Type has been deleted successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Type was not deleted.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    deleteEventTopic(eventTopic) {
+      this.set('isLoading', true);
+      eventTopic.destroyRecord()
+        .then(() => {
+          this.notify.success(this.get('l10n').t('Event Topic has been deleted successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Topic was not deleted.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    deleteEventSubTopic(eventSubTopic) {
+      this.set('isLoading', true);
+      eventSubTopic.destroyRecord()
+        .then(() => {
+          this.notify.success(this.get('l10n').t('Event Sub-Topic has been deleted successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Sub-Topic was not deleted.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    addEventType() {
+      this.set('isLoading', true);
+      this.get('eventType').save()
+        .then(() => {
+          this.set('isNewEventTypeModalOpen', false);
+          this.notify.success(this.get('l10n').t('Event Type has been added successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Type not saved.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    addEventTopic() {
+      this.set('isLoading', true);
+      this.get('eventTopic').save()
+        .then(() => {
+          this.set('isNewEventTopicModalOpen', false);
+          this.notify.success(this.get('l10n').t('Event Topic has been added successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Topic not saved.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    addEventSubTopic() {
+      this.set('isLoading', true);
+      let eventSubTopic;
+      eventSubTopic = this.get('eventSubTopic');
+      eventSubTopic.save()
+        .then(() => {
+          this.set('isNewEventTopicModalOpen', false);
+          this.notify.success(this.get('l10n').t('Event Sub Topic has been added successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Sub Topic not saved.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    async fetchSubTopics(getEventTopic) {
+      return {
```

Comment:
Why are you returning an object ? You can directly return subTopics right ?

Reply:
I forgot to remove this code. This is not used right now. Sorry. :sweat_smile: